lpc55_sign/src/signed_image.rs

```diff
@@ -24,366 +36,166 @@ fn get_pad(val: usize) -> usize {
     }
 }
 
-#[derive(Clone, Debug, Deserialize)]
-#[serde(rename_all = "kebab-case", deny_unknown_fields)]
-pub struct CertChain {
-    pub cert_paths: Vec<PathBuf>,
-    pub priv_key: Option<PathBuf>,
-}
-
-#[derive(Clone, Debug, Deserialize)]
-#[serde(rename_all = "kebab-case", deny_unknown_fields)]
-pub struct CfgFile {
-    pub certs: Vec<CertChain>,
-}
-
-fn validate_certs(certs: &[CertChain]) -> Result<()> {
-    if certs.len() > 4 {
-        return Err(anyhow!("Too many certificate chains, max is 4"));
-    }
-
-    let cnt = certs.iter().filter(|c| c.priv_key.is_some()).count();
-
-    if cnt != 1 {
-        return Err(anyhow!(
-            "Exactly one certificate chain should specify a private key for signing"
-        ));
+fn pad_roots(mut roots: Vec<Vec<u8>>) -> Result<Vec<Vec<u8>>> {
+    match roots.len().cmp(&4) {
+        Ordering::Equal => Ok(roots),
+        Ordering::Less => {
+            let empty = [vec![], vec![], vec![], vec![]];
+            roots.extend_from_slice(&empty[..4 - roots.len()]);
+            Ok(roots)
+        }
+        Ordering::Greater => bail!("Too many roots, max four"),
```

Comment:
```suggestion
    if roots.len() > 4 {
        bail!("Too many roots, max four");
    }
    roots.resize_with(4, Vec::new);
    Ok(roots)
```

Reply:
I ended up with that ackward match on `cmp` from clippy, but this is better all around. Thank you, fixed in de42dd9.